Littlefs/README.md

```diff
@@ -0,0 +1,126 @@
+
+Start the Python application before running the Azure Sphere application.
+
+**Azure Sphere High Level Application** 
+
+The high level application sets up the storage size (lines 24-27)
+
+```cpp
+// 4MB Storage.
+#define PAGE_SIZE     (256)
+#define SECTOR_SIZE   (16 * PAGE_SIZE)
+#define BLOCK_SIZE    (16 * SECTOR_SIZE)
+#define TOTAL_SIZE    (64 * BLOCK_SIZE)
+```
+
+The storage size information and pointers to read, program, erase, and sync are used to initialize Littlefs (lines 40-53) in main.c.
+
+The Azure Sphere high level application uses Curl to make HTTP Get/Post calls to read/write to Python virtual storage.
+
+You will need to modify the high level application app_manifest.json to provide the IP address of the PC running the Python application - modify **AllowedConnections** to include the PC host IP address, an example is below.
+
+```cpp
+{
+  "SchemaVersion": 1,
+  "Name": "LittleFs_HighLevelApp",
+  "ComponentId": "1689d8b2-c835-2e27-27ad-e894d6d15fa9",
+  "EntryPoint": "/bin/app",
+  "CmdArgs": [],
+  "Capabilities": {
+    "AllowedConnections": ["192.168.1.65"]
+  },
+  "ApplicationType": "Default"
+}
+```
+
+You will also need to set the IP address of the PC running the Python application in the projects CMakeLists.txt, modify `line 25` per the example below:
+
+```makefile
+# // TODO: define the PC HOST IP Address
+add_compile_definitions(PC_HOST_IP="192.168.1.65")
+```
+
+The application will initialize and format the storage, create a file, write some test data, rewind the file pointer, read the data and print to the debug console, and then close the file.
+
+When the application runs you should see the following output in the Visual Studio/Code debug window:
+
+```makefile
+Format and Mount
+Open File 'test.txt'
+Write to the file
+Seek to the start of the file
+Read from the file
+Read content = Test
+Close the file
+```
+
+### Curl memory tracing
+
+The application can also track Curl memory allocations, to enable this simply uncomment `line 22` in CMakeLists.txt
+
+```makefile
+# add_compile_definitions(ENABLE_CURL_MEMORY_TRACE)
+```
+
+Enabling Curl memory tracing combined with the [Gallery Heap Tracker project](https://github.com/Azure/azure-sphere-gallery/tree/main/HeapTracker) enables you to analyze memory usage within your application, specifically when using Curl API calls.
+
+With Curl memory tracking enabled you will see several memory tracking Log_Debug messages, this includes output from the following functions (malloc, free, calloc, strdup, realloc).
+
+## Project expectations
+
+* The code has been developed to support developers wanting to integrate a File System into their Azure Sphere project.
```

Comment:
the Project Expectations section should not focus on the aims of the project.  it should make it clear how production-ready/tested - or not - the code is.  For Gallery projects, unlike maintained samples, its OK for code to be "hacks" that are FYI and "at your own risk" and not be production-tested. 

As per the template:
```
Describe the state of the code. For example:

The code is not secure (for example, connection strings are included in the code).
The code is grossly inefficient (for example, loops that run unnecessarily because it uses the wrong breakout points).
The code does not run (for example, there are build errors).
The code may cause a problem if used in a production scenario.
The code has not been run through a complete security analysis (for example, there are static analysis failures).
```